pkg/transformers/namespace.go

```diff
@@ -31,13 +60,37 @@ func NewNamespaceTransformer(ns string) Transformer {
 	if len(ns) == 0 {
 		return NewNoOpTransformer()
 	}
-	return &namespaceTransformer{namespace: ns}
+
+	sort.Sort(ByPathConfig(namespacePathConfigs))
+
+	return &namespaceTransformer{
+		namespace:   ns,
+		pathConfigs: namespacePathConfigs,
+	}
 }
 
 // Transform adds the namespace.
 func (o *namespaceTransformer) Transform(m resmap.ResMap) error {
-	for _, res := range m {
-		res.SetNamespace(o.namespace)
+	for id := range m {
+		objMap := m[id].UnstructuredContent()
+
+		for _, path := range o.pathConfigs {
+			if !selectByGVK(id.Gvk(), path.GroupVersionKind) {
+				continue
+			}
+
+			if path.Skip {
+				break
```

Comment:
continue?

Reply:
No if we use continue we would just go to the next path which would eventually be:
 
```
	{
		Path:               []string{"metadata", "namespace"},
		CreateIfNotPresent: true,
	},
```
And the the namespace would end up getting set.